packages/core/src/components/dialog/multistepDialog.tsx

```diff
@@ -123,7 +123,14 @@ export class MultistepDialog extends AbstractPureComponent2<IMultistepDialogProp
         return (
             <div className={classNames(Classes.DIALOG_STEP_CONTAINER, { [Classes.ACTIVE]: hasBeenViewed })} key={index}>
                 <div className={Classes.DIALOG_STEP} onClick={this.handleClickDialogStep(index)}>
-                    <div className={Classes.DIALOG_STEP_ICON}>{stepNumber}</div>
+                    <div
+                        className={classNames(Classes.DIALOG_STEP_ICON, {
+                            [Classes.ACTIVE]: currentlySelected,
```

Comment:
The container div above is already `Classes.ACTIVE`. That should probably change to only be applied to the current step, and then you can use CSS to target the icon based on that and the new viewed class if necessary (see other comment).